src/screening.f90

```diff
@@ -127,6 +127,69 @@ subroutine calculate_Hilbert_weights(w_disc, w_cont, zeroplus, Hilbert_weights)
     end do
   end subroutine calculate_Hilbert_weights
 
+   subroutine Hilbert_transform(Reeps_T, Omegas, spec_eps_T)
+    !! Does Hilbert tranform of spectral head of bare polarizability 
+    !! Ref - EQ (4), R. Balito et. al. 
+    !!
+    !! Reeps_T Real part of bare polarizability
+    !! Omega Energy of excitation in the electron gas
+    !! spec_eps_T Spectral head of bare polarizability
+    !! 
+    
+    real(r64), allocatable, intent(out) :: Reeps_T(:)
+    real(r64), intent(in) :: Omegas(:)
+    real(r64), intent(in) :: spec_eps_T(:)
+    
+    ! Local variables
+    real(r64) :: sum1, sum2, diff, bh
+    integer(i64) :: nOmegas, kh, nh
+    ! Local constant
+    real(r64), parameter :: smalln = 1e-20  ! a number close to zero
+    
+    nOmegas = size(Omegas)
+    allocate(Reeps_T(nOmegas))
+
+    !
+    do kh = 1, nOmegas
+        sum1 = 0.0_r64
+        sum2 = 0.0_r64
+        do nh = 1, nOmegas - 1 - kh
+           bh = log((nh + 1.0_r64)/nh)
+           ! To avoid edge error in the upper limit
+           if ((kh + nh) < nOmegas) then
+              sum1 = sum1 - (1.0_r64 - (nh + 1.0_r64)*bh)*spec_eps_T(kh + nh) + &
+                          (1.0_r64 - nh*bh)*spec_eps_T(kh + nh + 1)
+           else 
+              ! For outside the upper boundary, assume the function = small number -> 0
+              sum1 = sum1 - (1.0_r64 - (nh + 1.0_r64)*bh)*spec_eps_T(kh + nh) + &
+                      (1.0_r64 - nh*bh)*smalln
+           end if
+        end do
+
+        do nh = 1, kh - 1
+           bh = log((nh + 1.0_r64)/nh)
+           ! To avoid edge error in the lower limit
+           if ((kh - nh) > 1) then
```

Comment:
Why is there a space after `if`? I don't think this done anywhere else in this code base.